pytest/blueprints/constructables/test_higher_order_iterator_constructables.py

```diff
@@ -81,6 +122,49 @@ def test_constructable(self, informed_iterators):
         # print(DatasetIteratorReportGenerator.generate_report(iterator_full))
 
 
+class TestInMemoryDatasetIteratorConstructable(IteratorFixtures):
+    def test_constructable(self, informed_iterators):
```

Comment:
Ok, so now we test if the component can be **constructed**. I think in a second step we should also test the **functionality** of the constructed component as well. In the case of the inmemory dataset iterator, we would check if everything is loaded into memory correctly, for instance. We should discuss this. :)